File: backend/src/main/scala/microservices/content/objects/ListReviewsByResourcePlannerRequest.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ListReviewsByResourcePlannerRequest(
    userId: String,
    resourceType: String,
    resourceId: String
)
object ListReviewsByResourcePlannerRequest:
  given sourceEncoder: Encoder[ListReviewsByResourcePlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[ListReviewsByResourcePlannerRequest] = deriveDecoder

File: frontend/src/microservices/content/objects/ListReviewsByResourcePlannerRequest.ts
// ListReviewsByResourcePlannerRequest：content 域资源评论列表请求对象。

export type ListReviewsByResourcePlannerRequest = {
  userId: string
  resourceType: string
  resourceId: string
}

export const listReviewsByResourcePlannerRequestFromJson = (json: string): ListReviewsByResourcePlannerRequest =>
  JSON.parse(json) as ListReviewsByResourcePlannerRequest

export const listReviewsByResourcePlannerRequestToJson = (value: ListReviewsByResourcePlannerRequest): string =>
  JSON.stringify(value)
